src/vm-vcpu/src/vcpu/mod.rs

```diff
@@ -183,19 +180,16 @@ impl KvmVcpu {
         let mut sregs = self.vcpu_fd.get_sregs().map_err(Error::KvmIoctl)?;
 
         // Global descriptor tables.
-        let gdt_table: [u64; BOOT_GDT_MAX as usize] = [
-            gdt_entry(0, 0, 0),            // NULL
-            gdt_entry(0xa09b, 0, 0xfffff), // CODE
-            gdt_entry(0xc093, 0, 0xfffff), // DATA
-            gdt_entry(0x808b, 0, 0xfffff), // TSS
-        ];
+        let gdt_table = Gdt::default();
 
-        let code_seg = kvm_segment_from_gdt(gdt_table[1], 1);
-        let data_seg = kvm_segment_from_gdt(gdt_table[2], 2);
-        let tss_seg = kvm_segment_from_gdt(gdt_table[3], 3);
+        let code_seg = gdt_table[1].create_kvm_segment(1);
```

Comment:
I am wondering if it would make sense to have this method as part of `Gdt`, and avoid this way repeating the index. Looks like the `table_index` would always be the same as the index from the `Gdt` (or not?).

Basically something like:

```rust
impl Gdt {
    pub fn create_kvm_segment(&self, table_index: u8) -> kvm_segment {
       let entry = self.get(table_index as usize).ok_or(...);
       ...
    }
}
```

I think this way we could also validate in `create_kvm_segment` that the table_index is valid and return an error if not, as opposed to `gdt_table[x]` which would panic.

Reply:
I'll think about this a bit longer. The current abstraction is not ideal, but I am not sure that it helps to add another level of indirection by having this function part of the `Gdt` and having to validate indexes. I'd rather have an assert!(gdt_table.len((), 4) before doing these accesses to avoid any risk of running into a programming error.

I'll come back to this in a bit.

Reply:
I don't particularly like the fact that you have to be careful to pass the same index twice, and it seemed like a good opportunity to avoid this with the `Gdt` abstraction. We could also have an assert in `create_kvm_segment` for the index validation and then use self[table_index as usize]. But I'm fine anyway, it was just a suggestion.